feat: Hip3 Perps Home Screen

gambinish opened the pull request:
## **Description**

Introduces new home screen + Hip-3 support UI. Initially broken out from https://github.com/MetaMask/metamask-mobile/pull/21410 to make it easier to iterate on, and isolate codeowners.

More complete technical detail in original ticket. This includes the home page redesign with hip-3 support, along with [after hours UX](https://github.com/MetaMask/metamask-mobile/pull/21901).

Figma designs: https://www.figma.com/design/kdRiGFFmRrkbCJyKf4c4Re/Equity-perps-and-more?node-id=153-3116&p=f&t=AS2BitV7DlFSsEWX-0

## **Changelog**

CHANGELOG entry: Introduce new Perps Home screen UI

## **Related issues**

Fixes:

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

## **Screenshots/Recordings**


https://github.com/user-attachments/assets/db35b6dd-ba76-4b95-bca1-f043bc71e27e

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.



<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Implements a new Perps Home UI with search/sections and bulk actions, overhauls Market List with tabs/sorting/filtering, adds HIP-3 2× fee logic and market-hours UX, refactors close flows and hooks, and updates routes, tests, and i18n.
> 
> - **UI/Views**:
>   - New `PerpsHomeView` with positions/orders carousels, watchlist, navigation card, search bar, and bottom sheets for bulk actions.
>   - Added `PerpsCloseSummary` shared component for close flows; integrated into Close Position and Close All modals.
>   - Added bulk action modals: `PerpsCloseAllPositionsModal` and `PerpsCancelAllOrdersModal`.
> - **Market List Overhaul**:
>   - New components: `PerpsMarketList`, `PerpsMarketListHeader`, `PerpsMarketFiltersBar`, `PerpsMarketTypeSection`, `PerpsStocksCommoditiesDropdown/BottomSheet`.
>   - Tabs (All/Crypto/Stocks & Commodities), favorites filter, search, and sort presets (volume, price change, funding, OI).
>   - Filters zero-volume/fallback markets by default; optional inclusion via flag.
> - **Fees/Providers**:
>   - Add HIP-3 fee multiplier (2× base protocol rate) with `coin` propagated through fee calculations.
>   - Updated `HyperLiquidProvider` and controller/types; extended tests.
> - **Market Hours**:
>   - `PerpsMarketHoursBanner` and tooltip with `marketHours` utils; integrated into Market Details.
> - **Hooks**:
>   - New hooks: `usePerpsHomeData`, `usePerpsMarketListView`, `usePerpsSearch`, `usePerpsSorting`, `usePerpsCloseAllCalculations`, `usePerpsCancelAllOrders`, `usePerpsCloseAllPositions`.
>   - `usePerpsOrderForm`: prioritize leverage (route > existing position > saved config > default) and sync on position load.
>   - `usePerpsLiveFills` now returns `{fills,isInitialLoading}`.
> - **Navigation/Routes**:
>   - Home view wired as main Perps screen; added modal routes for bulk actions.
> - **Misc**:
>   - Numerous tests for new components/hooks; style and i18n updates (home labels, market-hours strings).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 5e569095398839348199eae36cebbc7482ae7174. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

abretonc7s committed:
feat: hyperliquid docs embedded for llm

abretonc7s committed:
feat: ui changes for hip3

abretonc7s committed:
working state with tpsl live

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/hip3

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
fix: unit tests

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/hip3

abretonc7s committed:
fix: bugbot

abretonc7s committed:
cleanup removing magic nums

abretonc7s committed:
Merge branch 'main' into perps/feat/hip3

abretonc7s committed:
fix: linting issues

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/hip3

abretonc7s committed:
Merge branch 'main' into perps/feat/hip3

abretonc7s committed:
feat: styling badges

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/hip3

abretonc7s committed:
cleanup

abretonc7s committed:
fix: bugbot comment

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
feat: re-enable hip3 debug page for testing

abretonc7s committed:
feat: cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
feat: feature flag improvement and controller init

abretonc7s committed:
cleanup

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/hip3

abretonc7s committed:
feat: pr commentS

abretonc7s committed:
fix: unit tests

abretonc7s committed:
feat: change default equityEnabled flag to true on dev only

abretonc7s committed:
fix: unit test

abretonc7s committed:
fix: unit tests

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/hip3

abretonc7s committed:
fix: sonarcloud issues

abretonc7s committed:
fix: linting

abretonc7s committed:
fix: sonarcloud

abretonc7s committed:
fix: sonarcloud

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
feat: cleanup

abretonc7s committed:
feat: tat-1784

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
coverage

abretonc7s committed:
coverage

abretonc7s committed:
feat: coverage

abretonc7s committed:
feat(perps): add home screen v2, market sorting, and batch operations

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
cleanup

abretonc7s committed:
fix: unit test

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/hip3

abretonc7s committed:
pr comment

abretonc7s committed:
feat: add error handling to perpsdex

abretonc7s committed:
fix: overfund dex to allow acreating new position over existing

abretonc7s committed:
merge main into branch

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
fix: cleanup

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/hip3

abretonc7s committed:
testing auto transfer

abretonc7s committed:
feat: hip3 abstraction support

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/hip3

abretonc7s committed:
fix: tpsl matching

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/hip3

abretonc7s committed:
merge main into branch

abretonc7s committed:
feat: utility funciton to determine maker status

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
merge hip3 into branch

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
feat(sentry): add searchable tags and context to Logger.error

abretonc7s committed:
feat: add unit tests

abretonc7s committed:
Merge branch 'main' into feat/sentry-tags

abretonc7s committed:
feat: closing all positions

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
fix: unit testS

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into feat/sentry-tags

abretonc7s committed:
fix: coverage

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
feat: refactor for unit test coverage

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
Merge branch 'main' into feat/sentry-tags

abretonc7s committed:
fix: unit tests

abretonc7s committed:
feat: handle legazy format

abretonc7s committed:
fix(perps): improve error handling and navigation safety

- Add Sentry logging for stream pause/resume failures
- Fix race condition using Promise-based guard in cancelOrders
- Add navigation.canGoBack() checks with Markets fallback
- Clone enabledDexs array to prevent external mutations
- Update test mocks to include canGoBack method

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
Merge branch 'main' into feat/sentry-tags

abretonc7s committed:
feat: unit test

abretonc7s committed:
merge main into branch

abretonc7s committed:
Merge branch 'main' into feat/sentry-tags

abretonc7s committed:
Merge branch 'feat/sentry-tags' into perps/feat/newhome

abretonc7s committed:
fix: resolve duplicate symbol display and invalid concurrency test

abretonc7s committed:
fix: unit test and linting

abretonc7s committed:
feat(perps): add watchlist feature

abretonc7s committed:
feat: watchlist support and unit tests

abretonc7s committed:
fix: unit tests

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
fix: unit tests

abretonc7s committed:
feat: compute points based on currentprice not entryprice

abretonc7s committed:
feat: styling

abretonc7s committed:
feat: perps tab card navigation on everytabs

abretonc7s committed:
fix: bugbot

abretonc7s committed:
fix: unit tests

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
feat: support leverage trading preferences

abretonc7s committed:
coverage

abretonc7s committed:
feat: coverage and disable pull to refresh

abretonc7s committed:
feat: add loading indicator while loading fees

abretonc7s committed:
fix: perps recent activity list

abretonc7s committed:
feat: trending sorting page

abretonc7s committed:
feat: persisting trending sort preferences

abretonc7s committed:
fix: unit tests

abretonc7s committed:
fix: merge main into branch

abretonc7s committed:
fix: unit test + linting

abretonc7s committed:
fix: coverage

abretonc7s committed:
merge main into branch

abretonc7s committed:
cleanup

abretonc7s committed:
styling

abretonc7s committed:
fix: unit tests

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
feat: improve caching access

abretonc7s committed:
feat: merge main into branch

abretonc7s committed:
feat: perps card nav under tab

abretonc7s committed:
feat; unit tests

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
docs: improvements

abretonc7s committed:
fix: linting

abretonc7s committed:
fix: unit test

abretonc7s committed:
feat: use e2e selectors for testids

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
fix: unit tests

abretonc7s committed:
shared styles to avoid duplication

abretonc7s committed:
feat: unit test

abretonc7s committed:
sonarcloud

abretonc7s committed:
sonarcloud

abretonc7s committed:
sonarcloud

abretonc7s committed:
cleanup

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
cleanup

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
merge main into branch

abretonc7s committed:
feat: cleanup

abretonc7s committed:
feat: optimize batch cancel close orders positions

abretonc7s committed:
feat: improved loaders in home screen

abretonc7s committed:
feat: perps home view styling with market list

abretonc7s committed:
feat: simplify perps route for consistency

abretonc7s committed:
feat: cleanup ui component

abretonc7s committed:
feat: improved ui refactor with watchlist toggle

abretonc7s committed:
styling

abretonc7s committed:
fix: prevent invalid points computation

abretonc7s committed:
fix: unit tests

abretonc7s committed:
fix: linting

abretonc7s committed:
feat: navigation via hooks

abretonc7s committed:
fix: bugbot

abretonc7s committed:
fix: unit test coverage

abretonc7s committed:
feat: improved leverage validation

abretonc7s committed:
fix: unit tests

abretonc7s committed:
fix: unit tests

abretonc7s committed:
fix: unit tests

abretonc7s committed:
fix: unit test coverage

abretonc7s committed:
fix: unit tests

abretonc7s committed:
fix: unit tests

abretonc7s committed:
fix: coverage

abretonc7s committed:
fix: sonarcloud

abretonc7s committed:
feat: use crypto instead of perps

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
fix: bugbot

abretonc7s committed:
feat: hide bottom nav on marketlist

abretonc7s committed:
feat: remove watchlist from sort fields

gambinish committed:
chore: Merge main, address conflicts

cortisiko committed:
test(predict): add cash out position e2e  (#21545)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

The purpose of this test is to increase predictions e2e coverage by
adding a cash out position test. Scenario 4 from Test Plan
## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds an end-to-end cash-out flow test with comprehensive Polymarket
API/RPC/order-book mocks and new testIDs to enable stable UI automation.
> 
> - **E2E Tests**:
> - Add `predict-cash-out.spec.ts` covering cash-out flow (balances
update, activity entry, position removal) and related page objects.
> - Update `predict-api-down.spec.ts` to use new Polymarket mocks path.
> - **Mocks (Polymarket)**:
> - Introduce detailed mocks: event details (dynamic by `eventId`),
current/resolved positions (with optional winnings), order book per
`token_id`, activity, UpNL, and complete setup helpers.
> - Add post-action helpers: `POLYMARKET_POST_CASH_OUT_MOCKS`,
`POLYMARKET_FORCE_BALANCE_REFRESH_MOCKS`,
`POLYMARKET_REMOVE_CASHED_OUT_POSITION_MOCKS`, and post-claim mocks.
> - New constants for USDC balances (initial/post-claim/post-cash-out)
and addresses; track dynamic USDC balance in RPC mocks.
> - Remove legacy `polymarket-mocks.ts` and split into organized files.
> - **General Mocks**:
> - Expand price API defaults and add `polygon-rpc.com` default RPC
mock.
> - **Selectors & Pages**:
> - Add Predict Activity Details selectors and page; extend
ActivitiesView with predictions tab and helpers; refine predictions
selectors (stable position card IDs).
> - **App UI (testIDs)**:
> - Add `testID`s to `PredictActivityDetail` (back, title, amount,
container).
>   - Add `testID` to market details cash-out Button.
>   - Adjust position components’ `testID`s for stable selection.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
02b594feb62b7327329904d871dc6cde55edb7ac. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

gambinish committed:
Merge branch 'main' into perps/feat/newhome

tommasini committed:
chore: revert hardcoded fingerprint (#21617)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Replace hardcoded fingerprint with `yarn fingerprint:generate` in
`.github/workflows/build-android-e2e.yml` to compute the cache key
dynamically.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
47f5d9a1da6406c6111a8f0d0736c848e40a7f1e. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

---------

Co-authored-by: metamaskbot <metamaskbot@users.noreply.github.com>

bsgrigorov committed:
fix(INFRA-3066): path based rebuilds (#21747)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR fixes a critical bug in the GitHub Actions workflow that was
preventing native code changes from triggering E2E test rebuilds.

**Problem:** The negation pattern `!scripts/build.sh` in the paths
filter was breaking the entire filtering logic, causing the
dorny/paths-filter action to incorrectly treat all files (including
native Android and iOS code) as ignored. This meant that PRs with only
native code changes would not trigger app rebuilds, potentially running
E2E tests against stale builds.

**Solution:**
1. Moved `scripts/build.sh` from the `ignore` section (with problematic
negation) to the `shared` section
2. Added comprehensive documentation explaining the intended filter
logic
3. Ensured native code changes (`android/**`, `ios/**`) properly trigger
platform-specific rebuilds

This fix ensures the CI workflow correctly identifies when rebuilds are
needed while maintaining the intended optimization of avoiding
unnecessary rebuilds for documentation-only changes.

## **Changelog**

CHANGELOG entry: null

## **Related issues**

## **Manual testing steps**

1. Create a test PR with only Android native code changes (e.g., modify
a file in `android/app/src/`)
2. Verify the `needs_e2e_build` job outputs `android_changed: true` and
`builds: true`
3. Create a test PR with only iOS native code changes (e.g., modify a
file in `ios/`)
4. Verify the `needs_e2e_build` job outputs `ios_changed: true` and
`builds: true`
5. Create a test PR with only documentation changes (e.g., modify
`README.md`)
6. Verify the `needs_e2e_build` job outputs `builds: false` (no
unnecessary rebuilds)
7. Create a test PR that modifies `scripts/build.sh`
8. Verify it triggers rebuilds for both platforms (`android_changed:
true`, `ios_changed: true`)

## **Screenshots/Recordings**

### **Before**

Native code changes were incorrectly flagged as ignored:

### **After**

Native code changes now properly trigger the appropriate platform
builds.

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I've applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Refines path filters in `needs-e2e-build.yml` by moving
`scripts/build.sh` to `shared`, clarifying ignore rules, and adding
clear documentation to ensure correct Android/iOS rebuild triggers.
> 
> - **CI Workflow (`.github/workflows/needs-e2e-build.yml`)**:
>   - **Path filtering**:
> - Move `scripts/build.sh` from `ignore` to `shared` to trigger
rebuilds on changes.
> - Keep native paths `android/**` and `ios/**` as platform-specific
triggers; group common/dependency files under `shared`.
> - Maintain `ignore` for docs/workflow and utility scripts; refine
patterns and comments for clarity.
>     - Preserve `catch_all` fallback and output listing configuration.
> - **Documentation**: Add detailed comments explaining filter intent
and key points.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
ddc5d0b0b907563ded27b8f86f13d4dd575fe479. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

abretonc7s committed:
cleanup

abretonc7s committed:
Merge branch 'main' into perps/feat/newhome

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
reset to main

abretonc7s committed:
reset to main

abretonc7s committed:
reset to main

abretonc7s committed:
reset to main

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome

abretonc7s committed:
feat(perps): core infrastructure and WebSocket architecture

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/core-infrastructure

abretonc7s committed:
cleanup

gambinish committed:
chore: Merge main, address conflicts

gambinish committed:
fix: Unformat unrelated Card file

gambinish committed:
fix: Lint tsc, add remaining types

gambinish committed:
chore: update PerpsMarketTabs.test.tsx

gambinish committed:
chore: Add PerpsMarketDetailsView.test.tsx

gambinish committed:
chore: Add PerpsTabView test

gambinish committed:
chore: Add useLiveOrders.test.ts

gambinish committed:
chore: Add updated navigation stack

gambinish committed:
Merge branch 'main' into perps/feat/core-infrastructure

gambinish committed:
chore: Merge main address conflicts

gambinish committed:
fix: Remove formatting diff

gambinish committed:
fix: formatting diff

gambinish committed:
chore: formatting diff

gambinish committed:
fix: formatting diff

gambinish committed:
fix: formatting diff

gambinish committed:
chore: Remove PerpsBottomTabBar

gambinish committed:
chore: Update styling of perps and equity perps markets

gambinish committed:
Merge branch 'main' into perps/feat/core-infrastructure

gambinish committed:
chore: Rename deprecated anonymous type from BaseController

gambinish committed:
Merge branch 'main' into perps/feat/core-infrastructure

gambinish committed:
Merge branch 'perps/feat/core-infrastructure' into perps/feat/newhome-nick

gambinish committed:
chore: Update home page sections, consolidate stocks and commodities

gambinish committed:
Merge branch 'perps/feat/newhome-nick' of github.com:MetaMask/metamask-mobile into perps/feat/newhome-nick

gambinish committed:
Merge branch 'main' into perps/feat/core-infrastructure

gambinish committed:
chore: Close all bottom sheet should overlay over PerpsHome, and navigate on close to PerpsHomeScreen

gambinish committed:
chore: Update bottom sheet ref

gambinish committed:
fix: Format usePerpsNavigation.test.ts

gambinish committed:
Merge branch 'main' into perps/feat/core-infrastructure

gambinish committed:
chore: Update MarketList to tabbed bar

gambinish committed:
chore: Add stock and commodity dropdown filter

gambinish committed:
chore: Tweak PerpsHome and MarketList header styles

gambinish committed:
chore: Update search bar behavior

gambinish committed:
chore: Update search bar behavior

gambinish committed:
chore: Rebase, address conflicts

gambinish committed:
Merge branch 'main' into perps/feat/core-infrastructure

gambinish committed:
Merge branch 'perps/feat/core-infrastructure' into perps/feat/newhome-nick

gambinish committed:
chore: Muted background color in cards

gambinish committed:
chore: Cleanup market list header, remove watchlist from market list view

gambinish committed:
Merge branch 'perps/feat/newhome-nick' of github.com:MetaMask/metamask-mobile into perps/feat/newhome-nick

gambinish committed:
chore: Update filter dropdown styles

nickewansmith committed:
chore: adds after hours trading messaging and info sheets for stocks (#21901)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

For reference, here are the latest designs:
https://www.figma.com/design/kdRiGFFmRrkbCJyKf4c4Re/Equity-Perps?node-id=196-4742&t=z5jYIC1C1reQWxiK-1

This PR adds after hours trading messaging and live timer countdown to
end of session or next session. Weekends are also considered
after-hours. 9:30 am- 4:00pm Weekdays are open hours.

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry: Adds after hours messaging to the market details page

## **Related issues**

Fixes: n/a

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds equity market-hours banner and an info bottom sheet with live
open/close countdown, integrating new market-hours utilities and tests.
> 
> - **Perps UI (Market Details)**:
> - Integrate `PerpsMarketHoursBanner` into `PerpsMarketDetailsView.tsx`
with info button opening `PerpsBottomSheetTooltip` using
`market_hours`/`after_hours_trading`.
> - Add state/handlers for market-hours tooltip and dynamic content key
via `getMarketHoursStatus()`.
> - **Tooltip System**:
> - Extend `PerpsBottomSheetTooltip` to support custom-header content
keys and new keys: `market_hours`, `after_hours_trading`.
> - Add `MarketHoursContent` (icon, title, body, and live
minute-by-minute countdown) and register in `contentRegistry`.
> - **Components**:
> - New `PerpsMarketHoursBanner` component (equity-only) showing
contextual messages and info action.
> - **Utilities**:
> - Add `utils/marketHours.ts` with EST-aware market open/close logic,
next transition, and `formatCountdown`; helper `isEquityAsset`.
> - **Tests**:
> - Add unit tests for `MarketHoursContent`, `PerpsMarketHoursBanner`,
and `marketHours` utilities (incl. edge cases and rollovers).
> - **Selectors & i18n**:
>   - Add E2E selectors for banner and tooltip.
> - Add English strings for banner and tooltips (titles, subtitles,
countdown labels).
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
5860a4edea2df8464af582c6b9f563bcf179e8d0. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

gambinish committed:
chore: Polish styles on after hours UX

gambinish committed:
chore: Tweak market list header styles

gambinish committed:
chore: Cleanup styles on market list badges

gambinish committed:
chore: Align styles between PerpsHomeHeader and PerpsMarketListHeader

gambinish committed:
chore: Header padding and i18n

gambinish committed:
fix: GOLD not showing up in lists

gambinish committed:
chore: Remove weird circleX icon in searchbar

gambinish committed:
chore: Reabse main

gambinish committed:
fix: Format

gambinish committed:
chore: Dedupe state variables

gambinish committed:
fix: Update unit tests

gambinish committed:
Merge branch 'main' into perps/feat/newhome-nick

nickewansmith commented:
A few things I noticed:


When clicking on the market list filter tab for stocks the tabs switch back and forth intermittently

When selecting the stocks list the stocks filter is not selected

When viewing the market detail of a stock there is a large gap at the bottom

abretonc7s committed:
Merge branch 'main' into perps/feat/newhome-nick

abretonc7s committed:
linting

abretonc7s committed:
Merge branch 'main' into perps/feat/newhome-nick

abretonc7s committed:
feat(perps): show hip-3 markets without volume in dev (#21986)

## **Description**

This PR fixes an issue where HIP-3 markets (stocks and commodities) with
$0.00 volume were incorrectly displayed in the Market List View and Home
View.

**What was the problem?**
- Markets with zero or invalid volume data were showing up in the Stocks
and Commodities tabs
- This was inconsistent with how crypto perps markets behave (they
correctly hide zero-volume markets)

**What is the solution?**
- Zero-volume markets are now filtered out consistently across all
market types
- Filtering logic moved to the data layer for consistency
- Both Market List View and Home View now hide markets with $0.00 volume
by default

## **Changelog**

CHANGELOG entry: null

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/TAT-1957

## **Manual testing steps**

```gherkin
Feature: Hide zero-volume HIP-3 markets

  Scenario: user views Market List View stocks tab
    Given app is running and Perps feature is enabled

    When user navigates to Market List View
    And user selects the "Stocks" tab
    Then user should only see stocks with valid volume data
    And user should NOT see any markets displaying "$0.00" volume

  Scenario: user views Market List View commodities tab
    Given app is running and Perps feature is enabled

    When user navigates to Market List View
    And user selects the "Commodities" tab
    Then user should only see commodities with valid volume data
    And user should NOT see any markets displaying "$0.00" volume

  Scenario: user views Perps Home View
    Given app is running and Perps feature is enabled

    When user navigates to Perps Home View
    Then user should only see markets with valid volume data in all carousels
    And user should NOT see any markets displaying "$0.00" volume
```

## **Screenshots/Recordings**

### **Before**

Markets with $0.00 volume were visible in the Stocks tab and other HIP-3
market tabs.

### **After**

Only markets with valid volume data are displayed. Zero-volume markets
are filtered out consistently.

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I've applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Overhauls Perps Markets and Home with searchable/sortable tabs,
consistent zero‑volume filtering, bulk Close/Cancel All flows, market
hours banner/tooltip, refined leverage priority, and extensive hook/test
updates.
> 
> - **UI/Navigation**:
> - **Home**: New `PerpsHomeView` with positions/orders sections,
watchlist, market type sections, navigation card, and bottom sheets for
Close All/Cancel All.
> - **Market List**: Reworked `PerpsMarketListView` with tabs
(All/Crypto/Stocks & Commodities), search header, sorting/filters, and
reusable `PerpsMarketList`/row components.
> - **Bulk Actions**: Added
`PerpsCloseAllPositionsView`/`PerpsCancelAllOrdersView` and modal
variants; shared `PerpsCloseSummary`.
> - **Market Hours**: Added `PerpsMarketHoursBanner` and bottom-sheet
tooltip content.
> - **Hooks/Logic**:
> - **Data**: New `usePerpsHomeData`, `usePerpsMarketListView`,
`usePerpsSearch`, `usePerpsSorting`; consistent zero‑volume filtering
moved to `usePerpsMarkets` (opt‑in via `showZeroVolume`).
> - **Bulk Ops**: `usePerpsCloseAllCalculations`,
`usePerpsCloseAllPositions`, `usePerpsCancelAllOrders` for fees/points
and execution.
> - **Streams**: `usePerpsLiveFills` now returns `{ fills,
isInitialLoading }`.
> - **Orders**: Leverage priority: nav param > existing position > saved
config > default; validation accepts `existingPositionLeverage`.
> - **Routing/Styles/Locale**:
> - Wired Home and modal routes; various style tweaks; added i18n for
market hours.
> - **Tests**: Extensive new unit tests across views, hooks, components,
and styles.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
c712e480db4628461b73880cda39a7b1d5caa800. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

abretonc7s committed:
Merge branch 'main' into perps/feat/newhome-nick

abretonc7s committed:
feat(perps): double fee for hip-3 assets (#21919)

## **Description**

Implement 2x fee multiplier for HIP-3 (HyperLiquid Improvement Proposal
3) assets as per HyperLiquid protocol specification.

**What changed:**
- HIP-3 assets (identified by `dex:SYMBOL` format like `xyz:TSLA`) now
have 2x base exchange fees
- Regular assets (e.g., `BTC`, `ETH`) are unaffected

**Why:**
HyperLiquid protocol charges double base fees for builder-deployed
perpetuals (HIP-3 assets):
- Regular taker: 0.045% -> HIP-3 taker: 0.090%
- Regular maker: 0.015% -> HIP-3 maker: 0.030%

**How:**
Applied 2x multiplier to all three fee rate sources:
1. Base rates (when no user-specific rates available)
2. Cached user rates (from fee cache)
3. API-fetched user rates (from HyperLiquid API)

## **Changelog**

CHANGELOG entry: Fixed HIP-3 perpetuals to display correct 2x exchange
fees

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/TAT-1950

## **Manual testing steps**

```gherkin
Feature: HIP-3 Fee Display

  Scenario: user views fees for HIP-3 asset
    Given user is on Perps trading screen

    When user selects a HIP-3 asset (e.g., "xyz:XYZ100")
    And user enters a trade amount
    Then fee modal should show provider fee ~0.086% (2x base rate)
    And total fee should be ~0.186% (provider + 0.1% MetaMask)

  Scenario: user views fees for regular asset
    Given user is on Perps trading screen

    When user selects a regular asset (e.g., "BTC")
    And user enters a trade amount
    Then fee modal should show provider fee ~0.043% (base rate)
    And total fee should be ~0.143% (provider + 0.1% MetaMask)
```

## **Screenshots/Recordings**

### **Before**
HIP-3 assets showed incorrect 0.043% provider fee (missing 2x
multiplier)
<img width="549" height="1061" alt="image"
src="https://github.com/user-attachments/assets/1394476a-e622-4e99-8889-a607cc93a609"
/>

### **After**
HIP-3 assets correctly show 0.086% provider fee (2x multiplier applied)
<img width="554" height="1151" alt="image"
src="https://github.com/user-attachments/assets/132e3609-ebd1-4a94-8778-a69783a445c8"
/>

## **Pre-merge author checklist**

- [x] I've followed MetaMask Contributor Docs and MetaMask Mobile Coding
Standards
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable (225 tests passing, added HIP-3
validation test)
- [x] I've documented my code using JSDoc format if applicable
- [x] I've applied the right labels on the PR

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed)
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots

---

## **Technical Details**

**Files Modified:**
- `app/components/UI/Perps/controllers/types/index.ts` - Added `coin`
parameter to `FeeCalculationParams`
- `app/components/UI/Perps/controllers/providers/HyperLiquidProvider.ts`
- Applied HIP-3 2x multiplier to all rate sources
- `app/components/UI/Perps/controllers/PerpsController.ts` - Updated
method signature
- `app/components/UI/Perps/hooks/usePerpsOrderFees.ts` - Pass `coin`
parameter
- `app/components/UI/Perps/hooks/usePerpsCloseAllCalculations.ts` - Pass
`coin` parameter
- Test files - Updated all tests to pass `coin` parameter

**Test Coverage:**
- All existing tests passing (225/225)
- Added `hip3-fee-validation.test.ts` with 8 validation tests
- Verified HIP-3 detection logic
- Verified 2x multiplier application

**Key Implementation:**
```typescript
// Detect HIP-3 assets by colon separator
const { dex } = parseAssetName(coin); // "xyz:TSLA" -> dex="xyz"
const isHip3Asset = dex !== null;

// Apply 2x multiplier to all rate sources
if (isHip3Asset) {
  feeRate *= 2;
}
```

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Implements a 2× fee multiplier for HIP-3 (`dex:SYMBOL`) assets and
updates fee calculation to require `coin`, applying the multiplier
across base, cached, and API-fetched rates with comprehensive test
updates.
> 
> - **Fees/HIP-3**:
> - Add `HIP3_FEE_CONFIG` with `FEE_MULTIPLIER: 2` in
`constants/hyperLiquidConfig.ts`.
> - Apply 2× multiplier for HIP-3 assets (detected via
`parseAssetName(coin)`) in `HyperLiquidProvider.calculateFees()` for:
>     - Base rates
>     - Cached user rates
>     - API-fetched user rates
> - **API/Types**:
> - Make `coin` required in `FeeCalculationParams`
(`controllers/types`).
> - Update `PerpsController.calculateFees` signature to accept
`FeeCalculationParams` and forward to provider.
> - **Hooks/Usage**:
> - Pass `coin` to fee calc in `usePerpsOrderFees` and
`usePerpsCloseAllCalculations`.
> - **Tests**:
>   - Update all fee tests to include `coin`.
> - Add tests verifying 2× HIP-3 fees for taker/maker in
`HyperLiquidProvider.test.ts`.
> - Minor test cleanups (remove unused placeholders, add
`isFeeCacheValid` edge case).
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
d096fa972aa741f9082d7743de90783b760d89ab. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

abretonc7s committed:
fix: unit tests

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/newhome-nick

abretonc7s committed:
fix: unit tests
